docs/pulses.ipynb

```diff
@@ -0,0 +1,251 @@
+    "We then need to define a pulse duration (start and end times) and a wavelength range.\n",
+    "\n",
+    "To create a pulse with 1M neutrons, uniformly distributed in the ranges of 1-3 ms for birth times,\n",
+    "and 1-10 Å for wavelengths, we write:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2699065-309c-439c-bcda-bf15d20edd0b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pulse = tof.Pulse(\n",
+    "    neutrons=1_000_000, tmin=1.0 * ms, tmax=3.0 * ms, lmin=1.0 * AA, lmax=10.0 * AA\n",
+    ")\n",
+    "pulse"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1bf9e593-fb94-4174-b4f1-7cab27b7cf49",
+   "metadata": {},
+   "source": [
+    "To inspect the data in the pulse, we can either look at it `.birth_times` or `.wavelengths` members"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e16ae03f-0bba-4c3a-8ccc-8ebbd8d9dc71",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pulse.wavelengths"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "75969bc3-9d28-4aa0-9f0b-7a6eafb3ed93",
+   "metadata": {},
+   "source": [
+    "or we can plot it with"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "49fec301-52a4-45a7-9438-b3f736e50b9f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pulse.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b1a5ab7b-bd10-4c1a-adb9-c3a9be12c343",
+   "metadata": {},
+   "source": [
+    "## Specifying time and wavelength distributions\n",
+    "\n",
+    "Pulses at neutron facilities are never flat.\n",
+    "To change the time and wavelength distributions of the neutrons in your pulse,\n",
+    "you can specify arrays of probabilities.\n",
+    "The array values represent the probabilities, while the associated coordinate represent the values to be sampled from.\n",
+    "\n",
+    "As an example, we create a triangular distribution for the neutron birth times,\n",
+    "and a distribution with only 4 possible values for the neutron wavelengths."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b1b80e2e-db09-4e7d-802b-e4e3bdeee341",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v = np.arange(30.0)\n",
+    "p_time = sc.DataArray(\n",
+    "    data=sc.array(dims=['time'], values=np.concatenate([v, v[::-1]])),\n",
+    "    coords={'time': sc.linspace('time', 0.1, 6.0, len(v) * 2, unit='ms')},\n",
+    ")\n",
+    "p_wav = sc.DataArray(\n",
+    "    data=sc.array(dims=['wavelength'], values=[1.0, 2.0, 3.0, 4.0]),\n",
+    "    coords={\n",
+    "        'wavelength': sc.array(\n",
+    "            dims=['wavelength'], values=[1.0, 2.0, 3.0, 4.0], unit='angstrom'\n",
+    "        )\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "pulse = tof.Pulse.from_distribution(neutrons=200_000, p_time=p_time, p_wav=p_wav)\n",
```

Comment:
Not sure if you explain it elsewhere, but it is not obvious how the two distributions interact. Are the simply "broadcast" against each other, i.e., this is not a full 2-D definition of an arbitrary pulse?